src/m365/purview/commands/retentioneventtype/retentioneventtype-add.spec.ts

```diff
@@ -0,0 +1,126 @@
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+import { accessToken } from '../../../../utils/accessToken';
+const command: Command = require('./retentioneventtype-add');
+
+describe(commands.RETENTIONEVENTTYPE_ADD, () => {
+  const displayName = 'Contract Expiry';
+  const description = 'A retention event type description';
+
+  //#region Mocked Responses
+  const requestResponse = {
+    displayName: displayName,
+    description: description,
+    createdDateTime: "2022-12-21T09:28:37Z",
+    lastModifiedDateTime: "2022-12-21T09:28:37Z",
+    id: "f7e05955-210b-4a8e-a5de-3c64cfa6d9be",
+    createdBy: {
+      user: {
+        id: null,
+        displayName: "John Doe"
+      }
+    },
+    lastModifiedBy: {
+      user: {
+        id: null,
+        displayName: "John Doe"
+      }
+    }
+  };
+  //#endregion
+
+  let atStub: sinon.SinonStub;
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    auth.service.accessTokens[(command as any).resource] = {
+      accessToken: 'abc',
+      expiresOn: new Date()
+    };
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any).items = [];
+    atStub = sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(false);
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post,
+      accessToken.isAppOnlyAccessToken
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.RETENTIONEVENTTYPE_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds retention event type', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/security/triggerTypes/retentionEventTypes`) {
+        return requestResponse;
+      }
+
+      return 'Invalid Request';
+    });
+
+    await command.action(logger, { options: { displayName: displayName } });
+    assert(loggerLogSpy.calledWith(requestResponse));
+  });
+
+  it('throws an error when we execute the command using application permissions', async () => {
+    atStub.restore();
```

Comment:
Just like the other PR: could you change this to `sinonUtil.restore(accessToken.isAppOnlyAccessToken)`?